src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3001,18 +3002,78 @@ namespace winrt::TerminalApp::implementation
         ShowWindowChanged.raise(*this, args);
     }
 
-    winrt::fire_and_forget TerminalPage::_SearchMissingCommandHandler(const IInspectable /*sender*/, const Microsoft::Terminal::Control::SearchMissingCommandEventArgs args)
+    Windows::Foundation::IAsyncOperation<IVectorView<MatchResult>> TerminalPage::_FindPackageAsync(hstring query)
     {
-        assert(!Dispatcher().HasThreadAccess());
+        const PackageManager packageManager = WindowsPackageManagerFactory::CreatePackageManager();
+        PackageCatalogReference catalogRef{
+            packageManager.GetPredefinedPackageCatalog(PredefinedPackageCatalog::OpenWindowsCatalog)
+        };
+        catalogRef.PackageCatalogBackgroundUpdateInterval(std::chrono::hours(24));
+
+        ConnectResult connectResult = catalogRef.Connect();
+        for (int retries = 0; connectResult.Status() != ConnectResultStatus::Ok && retries <= 3; ++retries)
+        {
+            if (retries == 3)
+            {
+                co_return nullptr;
+            }
+            connectResult = catalogRef.Connect();
+        }
```

Comment:
The retries check in the loop condition and inside the loop seem redundant.

You can deduplicate the two connect calls, by doing something like this:
```cpp
ConnectResult connectResult{ nullptr };

for (int retries = 0;;)
{
    connectResult = catalogRef.Connect();

    // Success? Continue with the remaining function.
    if (connectResult.Status() == ConnectResultStatus::Ok)
    {
        break;
    }

    // 3rd time this has failed? Return early.
    if (++retries == 3)
    {
        co_return nullptr;
    }
}
```